hypothesis-python/tests/nocover/test_precise_shrinking.py

```diff
@@ -0,0 +1,259 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Copyright the Hypothesis Authors.
+# Individual contributors are listed in AUTHORS.rst and the git log.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
```

Comment:
could we add a top-level comment defining "precise" and "sloppy" shrinks in this file? Since the tests are explicitly for this distinction and I hadn't run into the distinction elsewhere in the codebase.

Reply:
Yeah, definitely. 

It's funny, I thought this was a distinction I'd written down somewhere, but apparently it only lives in my head. I could have *sworn* it was in the paper about the Hypothesis shrinker, but I think that might have been an earlier draft because it's certainly not in the published one.